src/commands/logs.ts

```diff
@@ -155,78 +49,188 @@ export default async (cmd: { json: boolean; open: boolean; setup: boolean; watch
+    } else {
+      const data = {
+        textPayload,
+        // chokes on unmatched json payloads
+        // jsonPayload: jsonPayload ? jsonPayload.fields.message.stringValue : '',
+        jsonPayload: jsonPayload ? JSON.stringify(jsonPayload).substr(0, 255) : '',
+        protoPayload,
+      };
+      payloadData = data.textPayload || data.jsonPayload || data.protoPayload || ERROR.PAYLOAD_UNKNOWN;
+      if (payloadData && payloadData['@type'] === 'type.googleapis.com/google.cloud.audit.AuditLog') {
+        payloadData = LOG.STACKDRIVER_SETUP;
+        functionName = padEnd(protoPayload.methodName, 15);
+      }
+      if (payloadData && typeof payloadData === 'string') {
+        payloadData = padEnd(payloadData, 20);
+      }
+    }
+    // tslint:disable-next-line:no-any
+    const coloredStringMap: any = {
+      ERROR: chalk.red(severity),
+      INFO: chalk.cyan(severity),
+      DEBUG: chalk.green(severity), // includes timeEnd
+      NOTICE: chalk.magenta(severity),
+      WARNING: chalk.yellow(severity),
+    };
+    let coloredSeverity: string = coloredStringMap[severity] || severity;
+    coloredSeverity = padEnd(String(coloredSeverity), 20);
+    // If we haven't logged this entry before, log it and mark the cache.
+    if (!logEntryCache[insertId]) {
+      console.log(`${coloredSeverity} ${timestamp} ${functionName} ${payloadData}`);
+      logEntryCache[insertId] = true;
+    }
+  }
+}
+
+export async function setupLogs(): Promise<string> {
+  let projectId: string;
+  const promise = new Promise<string>((resolve, reject) => {
+    getProjectSettings().then(projectSettings => {
+      console.log(`${LOG.OPEN_LINK(LOG.SCRIPT_LINK(projectSettings.scriptId))}\n`);
+      console.log(`${LOG.GET_PROJECT_ID_INSTRUCTIONS}\n`);
+      prompt([
+        {
+          type: 'input',
+          name: 'projectId',
+          message: `${LOG.ASK_PROJECT_ID}`,
+        },
+      ])
+        // tslint:disable-next-line:no-any
+        .then((answers: any) => {
+          projectId = answers.projectId;
+          const dotfile = DOTFILE.PROJECT();
+          if (!dotfile) return reject(logError(null, ERROR.SETTINGS_DNE));
+          dotfile
+            .read()
+            .then((settings: ProjectSettings) => {
+              if (!settings.scriptId) logError(ERROR.SCRIPT_ID_DNE);
+              dotfile.write(Object.assign(settings, { projectId }));
+              resolve(projectId);
+            })
+            .catch((err: object) => {
+              reject(logError(err));
+            });
+        })
+        .catch((err: Error) => {
+          reject(console.log(err));
+        });
+    });
+  });
+  promise.catch(err => {
+    logError(err);
+    spinner.stop(true);
+  });
+  return promise;
+}
+
+/**
+ * Fetches the logs and prints the to the user.
+ * @param startDate {Date?} Get logs from this date to now.
+ */
+export async function fetchAndPrintLogs(formatJson: boolean, projectId?: string, startDate?: Date) {
```

Comment:
Now takes `formatJson`, and also now takes optional `projectId` (which was previously accessible in main function)